### PR TITLE
Increase phpstan level

### PR DESCRIPTION
**Pull request by core23**
## Subject

Fix minor stan issues.

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -1,5 +1,5 @@
 parameters:
```
Comment (VincentLanglet) [inline phpstan.neon.dist:1]:
```
includes:
    - phpstan-baseline.neon

parameters:
    level: 5
    bootstrapFiles:
        - vendor/bin/.phpunit/phpunit/vendor/autoload.php
    paths:
        - src
        - tests
    treatPhpDocTypesAsCertain: false
```

?

**Review by VincentLanglet — COMMENTED** [review]
You need the psalm-symfony plugin and the phpstan-symfony plugin IMHO

**Comment by core23** [comment]
Depends on #580 #585

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -1,3 +1,7 @@
 parameters:
 	ignoreErrors:
+		-
+			message: "#^Parameter \\#1 \\$metas of method Sonata\\\\SeoBundle\\\\Seo\\\\SeoPage\\:\\:setMetas\\(\\) expects array\\<string, array\\<string, mixed\\>\\>, array\\<string, string\\> given\\.$#"
```
Comment (VincentLanglet) [inline phpstan-baseline.neon]: I would say we can fix this

Comment (core23) [reply phpstan-baseline.neon]: No idea how to fix this. This is a test issue where we ignore the type hints.

Comment (VincentLanglet) [reply phpstan-baseline.neon]: I see, this is the call 
```
public function testInvalidMetas()
    {
        $this->expectException(\RuntimeException::class);

        $page = new SeoPage();
        $page->setMetas([
            'type' => 'not an array',
        ]);
    }
```

Indeed we cannot fix it, but on other project we ignored the error directly in the file with a psalm-suppress and phpstan-ignore https://github.com/sonata-project/SonataAdminBundle/blob/8a4e7f5210353c25b80614f2dec20bf4f7217ffe/tests/FieldDescription/TypeGuesserChainTest.php#L28

Kinda using the baseline as a TODO and this error will never be fixed.

**Comment by core23** [comment]
phpstan is failing in the ci, because the docker action is using an old version of the phpstan-symfony extension.

cc @OskarStark

**Comment by VincentLanglet** [comment]
> phpstan is failing in the ci, because the docker action is using an old version of the phpstan-symfony extension.
> 
> cc @OskarStark

You should remove this ci build https://github.com/sonata-project/SonataSeoBundle/blob/3.x/.github/workflows/qa.yml
There is already https://github.com/sonata-project/SonataSeoBundle/blob/3.x/.github/workflows/qa.yaml

**Comment by core23** [comment]
> > phpstan is failing in the ci, because the docker action is using an old version of the phpstan-symfony extension.
> > cc @OskarStark
> 
> You should remove this ci build https://github.com/sonata-project/SonataSeoBundle/blob/3.x/.github/workflows/qa.yml
> There is already https://github.com/sonata-project/SonataSeoBundle/blob/3.x/.github/workflows/qa.yaml

Good catch! 👍

**Review by VincentLanglet — DISMISSED** [review]

File: src/DependencyInjection/Compiler/ServiceCompilerPass.php [hunk 3]
```diff
@@ -33,9 +33,8 @@ public function process(ContainerBuilder $container): void
         $definition->addMethodCall('setHtmlAttributes', [$config['head']]);
         $definition->addMethodCall('setSeparator', [$config['separator']]);
 
-        if ($alias = $container->setAlias('sonata.seo.page', $config['default'])) {
-            $alias->setPublic(true);
-        }
+        $alias = $container->setAlias('sonata.seo.page', $config['default']);
```
Comment (jordisala1991) [inline src/DependencyInjection/Compiler/ServiceCompilerPass.php]: no need this otv, you can directly call setPublic

File: src/DependencyInjection/SonataSeoExtension.php [hunk 4]
```diff
@@ -103,6 +107,10 @@ private function configureSitemap(array $config, ContainerBuilder $container): v
 
     /**
      * Fix the sitemap configuration.
+     *
+     * @param array<mixed> $config
```
Comment (jordisala1991) [inline src/DependencyInjection/SonataSeoExtension.php]: mixed[]

File: tests/DependencyInjection/ConfigurationTest.php [hunk 5]
```diff
@@ -100,6 +103,11 @@ private function getDefaultConfiguration(): array
         ];
     }
 
+    /**
+     * @param array<array-key, mixed> $configs
+     *
+     * @return array<array-key, mixed>
```
Comment (jordisala1991) [inline tests/DependencyInjection/ConfigurationTest.php]: mixed[], but I would recommend you to look at how it is done on AdminBundle and copy the same here

File: tests/Seo/SeoPageTest.php [hunk 6]
```diff
@@ -55,11 +55,15 @@ public function testRemoveMeta(): void
         static::assertFalse($page->hasMeta('property', 'foo'));
     }
 
+    /**
+     * @psalm-suppress InvalidArgument
```
Comment (jordisala1991) [inline tests/Seo/SeoPageTest.php:59]: can we add a @see comment to explain why you have to supress?

**Review by jordisala1991 — CHANGES_REQUESTED** [review]

File: tests/Seo/SeoPageTest.php (hunk 6 above)
Comment (VincentLanglet) [reply tests/Seo/SeoPageTest.php:59]: Not really, this error is made on purpose to trigger the exception in the test.
We want to check that if we provide an invalid argument, an exception is thrown.

**Review by VincentLanglet — DISMISSED** [review]

File: src/DependencyInjection/SonataSeoExtension.php [hunk 7]
```diff
@@ -52,6 +52,8 @@ public function load(array $configs, ContainerBuilder $container): void
 
     /**
      * Configure the default seo page.
+     *
+     * @param array<mixed> $config
```
Comment (jordisala1991) [inline src/DependencyInjection/SonataSeoExtension.php]: mixed[] (there are more uses like this).

File: src/Seo/SeoPage.php [hunk 8]
```diff
@@ -272,7 +264,9 @@ public function getOEmbedLinks(): array
     }
 
     /**
-     * @param array|string $meta
+     * @param array<mixed>|string $meta
```
Comment (jordisala1991) [inline src/Seo/SeoPage.php]: string|mixed[]

File: src/DependencyInjection/Compiler/ServiceCompilerPass.php [hunk 9]
```diff
@@ -33,10 +33,7 @@ public function process(ContainerBuilder $container): void
         $definition->addMethodCall('setHtmlAttributes', [$config['head']]);
         $definition->addMethodCall('setSeparator', [$config['separator']]);
 
-        if ($alias = $container->setAlias('sonata.seo.page', $config['default'])) {
-            $alias->setPublic(true);
-        }
-
+        $container->setAlias('sonata.seo.page', $config['default'])->setPublic(true);
```
Comment (franmomu) [inline src/DependencyInjection/Compiler/ServiceCompilerPass.php:36]: Not related to this PR, but do we need this alias to be public?

**Review by franmomu — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**